src/Lean/Util/Path.lean

```diff
@@ -30,13 +30,19 @@ abbrev SearchPath := System.SearchPath
 namespace SearchPath
 
 /-- If the package of `mod` can be found in `sp`, return the path with extension
-`ext` (`lean` or `olean`) corresponding to `mod`. Otherwise, return `none.` -/
+`ext` (`lean` or `olean`) corresponding to `mod`. Otherwise, return `none`. Does
+not check whether the returned path exists. -/
 def findWithExt (sp : SearchPath) (ext : String) (mod : Name) : IO (Option FilePath) := do
   let pkg := mod.getRoot.toString
   let root? ← sp.findM? fun p =>
     (p / pkg).isDir <||> ((p / pkg).withExtension ext).pathExists
   return root?.map (modToFilePath · mod ext)
 
+/-- Like `findWithExt`, but ensures the returned path exists. -/
+def findModuleWithExt (sp : SearchPath) (ext : String) (mod : Name) : IO (Option FilePath) := do
```

Comment:
Let's base this one on `findWithExt` so that we're absolutely sure we always select the same package root